### PR TITLE
Added an `examples` test app activity to show Maps SDK + Nav SDK only combo 

### PR DESCRIPTION
**Pull request by langsmith**
## Description

Adds a new test app example to show how **only** the Navigation SDK can be used with the Maps SDK.

- [ ] I have added any issue links
- [x] I have added all related labels (`bug`, `feature`, `new API(s)`, `SEMVER`, etc.)
- [ ] I have added the appropriate milestone and project boards

### Goal

To show how the `1.0.0` Navigation SDK can be purely combined with the Maps SDK, with no usage of the `1.0.0` Navigation UI SDK. The test app's `BasicNavigationActivity` is definitely basic, but it still uses `NavigationMapboxMap`, which is a part of the Navigation UI SDK. Many other examples use `NavigationMapboxMap`.

### Implementation

The example added in this pr uses the:

- Maps SDK's `MapboxMap`
- `LocationComponent` to show the device location
- Navigation SDK to request a route from the device location to wherever the map is long clicked
- Maps SDK to show the Navigation SDK's route via a gradient styled Maps SDK `LineLayer`

## Screenshots or Gifs

ezgif com-resize (1)

## Testing

Please describe the manual tests that you ran to verify your changes

- [x] I have tested locally (including `SNAPSHOT` upstream dependencies if needed) through testapp/demo app and run all activities to avoid regressions
- [] I have tested via a test drive, or a simulation/mock location app
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have updated the `CHANGELOG` including this PR

### CODE REVIEW

File: examples/src/main/java/com/mapbox/navigation/examples/core/BasicNavSdkOnlyActivity.kt [hunk 1]
```diff
@@ -0,0 +1,241 @@
...
+    private fun enableLocationComponent() {
+        mapboxMap?.getStyle {
+            val locationComponentActivationOptions =
+                    LocationComponentActivationOptions.builder(this, it)
+                            .build()
+
+            // Get an instance of the LocationComponent and then adjust its settings
+            mapboxMap?.locationComponent?.apply {
+
+                // Activate the LocationComponent with options
```
Comment (Guardiola31337) [inline examples/src/main/java/com/mapbox/navigation/examples/core/BasicNavSdkOnlyActivity.kt:191]: Do we need this comment? I think method name is self-explanatory. Same applies to comments below 👇

File: examples/src/main/java/com/mapbox/navigation/examples/core/BasicNavSdkOnlyActivity.kt [hunk 2]
```diff
@@ -0,0 +1,236 @@
...
+     * Enable the Maps SDK's LocationComponent
+     */
+    private fun enableLocationComponent() {
+        mapboxMap?.getStyle {
+            val locationComponentActivationOptions =
+                    LocationComponentActivationOptions.builder(this, it)
+                            .build()
+
+            mapboxMap?.locationComponent?.apply {
+
```
Comment (Guardiola31337) [inline examples/src/main/java/com/mapbox/navigation/examples/core/BasicNavSdkOnlyActivity.kt:189]: NIT unnecessary line, same applies below 👇

File: examples/src/main/java/com/mapbox/navigation/examples/core/BasicNavSdkOnlyActivity.kt [hunk 3]
```diff
@@ -0,0 +1,236 @@
...
+            if (routes.isNotEmpty()) {
+                Snackbar.make(container, String.format(getString(R.string.steps_in_route),
+                        routes[0].legs()?.get(0)?.steps()?.size), LENGTH_SHORT).show()
+
+                // Update a gradient route LineLayer's source with the Maps SDK. This will
+                // visually add/update the line on the map. All of this is being done
+                // directly with Maps SDK code and NOT the Navigation UI SDK.
+                mapboxMap?.getStyle {
+                    val clickPointSource = it.getSourceAs<GeoJsonSource>("ROUTE_LINE_SOURCE_ID")
+
```
Comment (Guardiola31337) [inline examples/src/main/java/com/mapbox/navigation/examples/core/BasicNavSdkOnlyActivity.kt:158]: NIT unnecessary line

File: examples/src/main/java/com/mapbox/navigation/examples/core/BasicNavSdkOnlyActivity.kt (hunk 2 above)
Comment (langsmith) [reply examples/src/main/java/com/mapbox/navigation/examples/core/BasicNavSdkOnlyActivity.kt]: Fixed

File: examples/src/main/java/com/mapbox/navigation/examples/core/BasicNavSdkOnlyActivity.kt (hunk 3 above)
Comment (langsmith) [reply examples/src/main/java/com/mapbox/navigation/examples/core/BasicNavSdkOnlyActivity.kt]: Fixed

File: examples/src/main/java/com/mapbox/navigation/examples/core/BasicNavSdkOnlyActivity.kt (hunk 1 above)
Comment (langsmith) [reply examples/src/main/java/com/mapbox/navigation/examples/core/BasicNavSdkOnlyActivity.kt]: Not needed. Removed the comments.

**Pull request merged**